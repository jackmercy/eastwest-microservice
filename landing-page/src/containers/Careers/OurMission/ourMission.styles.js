import { createUseStyles } from 'react-jss';

import { rem } from '../../../utils/jss';
import BREAKPOINTS from '../../../constants/BREAKPOINTS';

export default createUseStyles({
  text: { fontSize: rem(24), lineHeight: rem(40) },
  image: {
    display: 'block',
    width: '100%',
    boxSizing: 'border-box',
    margin: [rem(48), 0],
  },

  [`@media only screen and (max-width: ${BREAKPOINTS.maxSm}px)`]: {
    text: { textAlign: 'center', fontSize: rem(16), lineHeight: rem(24) },
    image: { margin: [rem(32), 0] },
  },
});
