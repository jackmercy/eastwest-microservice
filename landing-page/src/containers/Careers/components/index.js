export { default as SectionTitle } from './SectionTitle';
export { default as MenuScrollableLayout } from './MenuScrollableLayout';
