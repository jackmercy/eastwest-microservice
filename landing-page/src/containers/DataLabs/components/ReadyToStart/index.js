import dynamic from 'next/dynamic';

export default dynamic(() => import('./ReadyToStart'), { ssr: false });
