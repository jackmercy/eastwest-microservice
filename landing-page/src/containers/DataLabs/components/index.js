export { default as Teaser } from './Teaser';
export { default as JoinUs } from './JoinUs';
export { default as Become } from './Become';
export { default as HowItWorks } from './HowItWorks';
export { default as ReadyToStart } from './ReadyToStart';
