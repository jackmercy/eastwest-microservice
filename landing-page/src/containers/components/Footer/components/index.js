export { default as Register } from './Register';
export { default as CopyRight } from './CopyRight';
export { default as SummaryInfo } from './SummaryInfo';
export { default as SectionTitle } from './SectionTitle';
export { default as SwitchLanguages } from './SwitchLanguages';
