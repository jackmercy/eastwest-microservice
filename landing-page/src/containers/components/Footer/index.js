import dynamic from 'next/dynamic';

export default dynamic(() => import('./Footer'), { ssr: false });
