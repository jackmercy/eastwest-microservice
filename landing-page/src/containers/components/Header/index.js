import dynamic from 'next/dynamic';

export default dynamic(() => import('./Header'), { ssr: false });
