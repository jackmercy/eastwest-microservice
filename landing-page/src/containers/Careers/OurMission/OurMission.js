import { memo } from 'react';
import PropTypes from 'prop-types';
import { useIntl } from 'react-intl';

import { SectionTitle } from '../components';

import useStyles from './ourMission.styles';

const OurMission = ({ id }) => {
  const intl = useIntl();
  const classes = useStyles();

  return (
    <>
      <SectionTitle id={id} translateId="CAREERS.OUR_MISSION.TITLE" />
      <p className={classes.text}>
        {intl.formatMessage({ id: 'CAREERS.OUR_MISSION.TEXT.01' })}
      </p>
      <img
        alt="our mission"
        className={classes.image}
        src="/static/images/careers/our-mission.png"
      />
      <p className={classes.text}>
        {intl.formatMessage({ id: 'CAREERS.OUR_MISSION.TEXT.02' })}
      </p>
    </>
  );
};
OurMission.propTypes = { id: PropTypes.string.isRequired };

export default memo(OurMission);
