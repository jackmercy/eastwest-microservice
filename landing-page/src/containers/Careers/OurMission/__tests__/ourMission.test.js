import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { SectionTitle } from '../../components';
import OurMission from '../OurMission';

describe('Testing Careers > OurMission', () => {
  let wrapper;
  const props = { id: 'our-mission' };

  beforeAll(() => {
    wrapper = mount(<OurMission {...props} />);
  });

  it('should render match snapshot', () => {
    const wrapper = shallow(<OurMission {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should have a SectionTitle', () => {
    expect(wrapper.find(SectionTitle)).toHaveLength(1);
  });
});
