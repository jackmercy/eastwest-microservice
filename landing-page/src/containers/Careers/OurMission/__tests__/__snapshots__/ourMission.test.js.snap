// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Careers > OurMission should render match snapshot 1`] = `
<Fragment>
  <Memo(SectionTitle)
    id="our-mission"
    translateId="CAREERS.OUR_MISSION.TITLE"
  />
  <p
    className="text-0-2-1"
  >
    We are on a mission to help everyone achieve product scalability. Our eyes are set beyond the world of engineering, and want to create an ecosytem anyone can use.
  </p>
  <img
    alt="our mission"
    className="image-0-2-2"
    src="/static/images/careers/our-mission.png"
  />
  <p
    className="text-0-2-1"
  >
    Our approach unites culture, process, tools and best practices that help them build quickly, release often and continuously Innovate. To keep scaling products going we’re going to need a few more critical thinkers. Join East West and help us build the future products that our users envision. The most rewarding thing is when you see people grow. To see them taking on more responsibility, setting their own goals and realizing what’s important to them. This happens at East West, every day.
  </p>
</Fragment>
`;
